### PR TITLE
551 cookie banner

### LINKED ISSUE #551: Cookie Banner

### PR DESCRIPTION
**Pull request by DilwoarH**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## What

- Add cookie banner
- Add cookie preference form in cookies page

## Why

We want to use Google analytics and other tools which requires users to consent to cookies.

## Related Tickets & Documents

- Ticket Link
- Related Issue #
- Closes #551 

## QA Instructions, Screenshots, Recordings

Screenshot 2024-10-25 at 11 50 28 pm

Screenshot 2024-10-25 at 11 51 19 pm
Screenshot 2024-10-25 at 11 50 45 pm
Screenshot 2024-10-25 at 11 50 37 pm
Screenshot 2024-10-25 at 11 50 51 pm

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Release Notes

- **New Features**
	- Introduced a cookie consent banner to manage user cookie preferences.
	- Added functionality for users to accept or reject cookies and update their preferences.
	- New HTML templates and styles for the cookie banner and cookie settings page.
	- Implemented a notification system to inform users of updated cookie preferences.
	- Enhanced accessibility features for the cookie banner.
	- Integrated cookie management functionality into the main application flow.

- **Bug Fixes**
	- Improved cookie management functionality for a better user experience.

- **Documentation**
	- Updated cookie-related pages to provide clearer information on cookie usage and settings.

### CODE REVIEW

**Review by GeorgeGoodall-GovUk — APPROVED** [review]
tested via preview link, looked over code, all looks good!

File: src/routes/cookies.js [hunk 1]
```diff
@@ -4,8 +4,17 @@ import nunjucks from 'nunjucks'
...
+    cookiePreferences: req.cookies.cookies_preferences_set,
+    showCookieUpdatedMessage: req.cookies.cookies_preferences_set_updated
+  })
   res.send(cookiesPage)
 })
 
+router.post('/update-preference', (req, res) => {
+  res.cookie('cookies_preferences_set', req.body.accept_cookies, { maxAge: 1000 * 60 * 60 * 24 * 365 })
+  res.cookie('cookies_preferences_set_updated', true, { maxAge: 1000 })
+  res.redirect('/cookies')
```
[the remark this answers is not in the document]
Comment (DilwoarH) [reply src/routes/cookies.js:23]: Don't you need sessions for flash to work?

**Comment by rosado** [comment]
Is the cookie banner tripping up the playwright tests?

**Review by rosado — DISMISSED** [review]
When I run this locally, I can't dismiss the banner.

**Comment by DilwoarH** [comment]
> When I run this locally, I can't dismiss the banner.

Interesting, I'll have a look

**Comment by DilwoarH** [comment]
> When I run this locally, I can't dismiss the banner.

Interesting, I'll have a look

**Comment by rosado** [comment]
Doh! correction: I forgot to bundle the client code

**Review by GeorgeGoodall-GovUk — APPROVED** [review]
lgtm

**Pull request merged**